src/main/java/uk/gov/companieshouse/api/accounts/controller/CompanyAccountController.java

```diff
@@ -20,8 +20,6 @@
     @PostMapping(value = "/transactions/{transactionId}/company-accounts",
             produces = MediaType.APPLICATION_JSON_VALUE)
     public ResponseEntity createCompanyAccount(@Valid @RequestBody CompanyAccount companyAccount) {
-        companyAccount = companyAccountService.createCompanyAccount(companyAccount);
-
-        return ResponseEntity.status(HttpStatus.CREATED).body(companyAccount);
+        return ResponseEntity.status(HttpStatus.CREATED).body(companyAccountService.save(companyAccount));
```

Comment:
If we are calling the generic `save` where would the validation fit in with this solution?

Reply:
Using a generic save will not effect the functionality of the validation implementation.

Reply:
Where would the validation fit in?

Reply:
This conversation should not be conducted via the PR process as it involves specific functionality of another piece of work, but if I was to take a stab at it without any investigation I would add an interface along the lines of 'public interface validation<T>'. Then add it into the AbstractServiceImpl class performing constructor injection to get a specific implementation in the same manner that we use for the GenericTransformer.

Reply:
I'm just making sure we don't commit to any structure that makes it harder for ourselves in future. If you believe it won't be a problem, then that's all I need to know.

Reply:
I know it's part of another piece of work but this wouldn't allow returning error HTTP responses if the validation fails or an error occurs inside companyAccountService.save when you have already set the return status code before calling save. One option would be to validate inputs before calling the service to handle the creation which would work with your existing code. I am assuming the global error handler catches all errors and returns the appropriate error code?

Reply:
This will not work for validation as is, the only error handling that is currently implemented correctly is the Bind error for the CompanyAccount class. All other errors would be in another PR we currently only return status 201 created. The purpose of this PR is to get the AbstractService in place so I can incrementally build on top of it.